src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -1965,6 +1968,27 @@ private bool LowerBoundInterfaceInference(TypeSymbol source, NamedTypeSymbol tar
             return true;
         }
 
+        internal static ImmutableArray<NamedTypeSymbol> ModuloNullabilityDifferences(IEnumerable<NamedTypeSymbol> interfaces, VarianceKind variance)
+        {
+            var dictionary = new Dictionary<NamedTypeSymbol, NamedTypeSymbol>(SymbolEqualityComparer.IgnoreNullableModifiersForReferenceTypes);
+
+            foreach (var @interface in interfaces)
+            {
+                if (dictionary.TryGetValue(@interface, out var found))
+                {
+                    dictionary.Remove(@interface);
+                    var merged = (NamedTypeSymbol)found.MergeNullability(@interface, variance);
+                    dictionary.Add(merged, merged);
+                }
+                else
+                {
+                    dictionary.Add(@interface, @interface);
+                }
+            }
+
+            return dictionary.Keys.ToImmutableArray();
```

Comment:
>return dictionary.Keys.ToImmutableArray(); [](start = 12, length = 42)

I think the most common case will be when nothing was merged, i.e. the dictionary the same count of elements as the original ```interfaces```. Consider returning the original ImmutableArray in that case.